### ISSUE TITLE
Strange behavior building an image

### ISSUE THREAD

**Issue by orelvis15** [fussybeaver/bollard#262]
Hello Goodnight!.
I am testing an example of the slightly modified lib it would be running like this.

```let mut build_image_args = HashMap::new();
    build_image_args.insert("version_url", versionNumber.as_str());
    build_image_args.insert("version_number", versionUrl.as_str());

    let build_image_options = BuildImageOptions {
        remote: dockerfileUrl.as_str(), //url to a regular Dockerfile
        buildargs: build_image_args,
        ..Default::default()
    };

    let mut image_build_stream = docker.build_image(build_image_options, None, None);

    while let Some(msg) = image_build_stream.next().await {
        println!("Message: {:?}", msg);
    }
```
[Dockerfile](https://raw.githubusercontent.com/orelvis15/kippy_data/master/servers/terraria/dockerfile)

When executing this code, an unusual behavior begins, first two images are created, one with the name **Ubuntu** and the other **None**, then containers are created up to a total of 21.

I leave the captures of how the final result looks.

<img width="1379" alt="Captura de Pantalla 2022-10-07 a la(s) 23 00 49" src="https://user-images.githubusercontent.com/19788313/194687229-cff14af7-2fc9-4175-8bef-6655236ffbfd.png">
<img width="1403" alt="Captura de Pantalla 2022-10-07 a la(s) 23 01 05" src="https://user-images.githubusercontent.com/19788313/194687241-04638d64-c588-4321-b190-5d527f30153b.png">

And these are the two images created.
<img width="1364" alt="Captura de Pantalla 2022-10-07 a la(s) 23 01 48" src="https://user-images.githubusercontent.com/19788313/194687258-8d03b9e5-a9aa-4933-8705-0a56efa837bb.png">
<img width="1360" alt="Captura de Pantalla 2022-10-07 a la(s) 23 01 58" src="https://user-images.githubusercontent.com/19788313/194687263-3658b40b-68a6-4a6d-885a-d6b40bafc718.png">

Any idea what my mistake would be?

**Comment by fussybeaver** [comment]
Could you explain your motivation or what you're trying to achieve? I don't see any problem yet...

**Comment by orelvis15** [comment]
Of course, what happens is that when you create an image using that dockerfile using the terminal, the corresponding image is created, including prove using the `std:: process` of rust and everything is ok, but when I do it using the lib, the behavior that it creates comes out two images and 21 containers which is not what I'm expecting, it should only create 1 single image without containers because I'm just building the image

**Comment by leon3s** [comment]
Hey @orelvis15 this is not an issue related to bollard. Container image are created using container. You have one container running for each command you set in your dockerfile. by default when you build using the docker cli it will remove them, when you use the API dirrectly you have to specify an option to delete them.
Look there https://docs.docker.com/engine/api/v1.41/#tag/Image/operation/ImageBuild
You will notice 2 options:
- rm Remove intermediate containers after a successful build.
- forcerm Always remove intermediate containers, even upon failure.
